tests/test_hooks/test_early_stopping_hook.py

```diff
@@ -0,0 +1,268 @@
+
+
+class ToyModel(BaseModel):
+
+    def __init__(self):
+        super().__init__()
+        self.linear = nn.Linear(2, 1)
+
+    def forward(self, inputs, data_sample, mode='tensor'):
+        labels = torch.stack(data_sample)
+        inputs = torch.stack(inputs)
+        outputs = self.linear(inputs)
+        if mode == 'tensor':
+            return outputs
+        elif mode == 'loss':
+            loss = (labels - outputs).sum()
+            outputs = dict(loss=loss)
+            return outputs
+        else:
+            return outputs
+
+
+class DummyDataset(Dataset):
+    METAINFO = dict()  # type: ignore
+    data = torch.randn(12, 2)
+    label = torch.ones(12)
+
+    @property
+    def metainfo(self):
+        return self.METAINFO
+
+    def __len__(self):
+        return self.data.size(0)
+
+    def __getitem__(self, index):
+        return dict(inputs=self.data[index], data_sample=self.label[index])
+
+
+class DummyMetric(BaseMetric):
+
+    default_prefix: str = 'test'
+
+    def __init__(self, length):
+        super().__init__()
+        self.length = length
+        self.best_idx = length // 2
+        self.cur_idx = 0
+        self.vals = [90, 91, 92, 88, 89, 90] * 2
+
+    def process(self, *args, **kwargs):
+        self.results.append(0)
+
+    def compute_metrics(self, *args, **kwargs):
+        acc = self.vals[self.cur_idx]
+        self.cur_idx += 1
+        return dict(acc=acc)
+
+
+def get_mock_runner():
+    runner = Mock()
+    runner.train_loop = Mock()
+    runner.train_loop.stop_training = False
+    return runner
+
+
+class TestEarlyStoppingHook(RunnerTestCase):
+
+    def setUp(self):
+        self.temp_dir = tempfile.TemporaryDirectory()
+
+    def tearDown(self):
+        # `FileHandler` should be closed in Windows, otherwise we cannot
+        # delete the temporary directory
+        logging.shutdown()
+        MMLogger._instance_dict.clear()
+        self.temp_dir.cleanup()
+
+    def test_init(self):
+
+        hook = EarlyStoppingHook(monitor='acc')
+        assert hook.rule == 'greater'
+        assert hook.best_score < 0
+
+        hook = EarlyStoppingHook(monitor='ACC')
+        assert hook.rule == 'greater'
+        assert hook.best_score < 0
+
+        hook = EarlyStoppingHook(monitor='ACC')
+        assert hook.rule == 'greater'
+        assert hook.best_score < 0
+
+        hook = EarlyStoppingHook(monitor='mAP_50')
+        assert hook.rule == 'greater'
+        assert hook.best_score < 0
+
+        hook = EarlyStoppingHook(monitor='loss')
+        assert hook.rule == 'less'
+        assert hook.best_score > 0
+
+        hook = EarlyStoppingHook(monitor='Loss')
+        assert hook.rule == 'less'
+        assert hook.best_score > 0
+
+        hook = EarlyStoppingHook(monitor='ce_loss')
+        assert hook.rule == 'less'
+        assert hook.best_score > 0
+
+        with pytest.raises(ValueError):
+            EarlyStoppingHook(monitor='recall')
+
+        with pytest.raises(ValueError):
```

Comment:
Suggest using `self.assertRaises`

Reply:
The rule `the world` reminds me of the comic [jojo](https://jojowiki.com/The_World)(one of my favorite comic :laughing: ). Just missing this comment If I misunderstand it :rofl: